src/chocolatey.resources/helpers/functions/Generate-BinFile.ps1

```diff
@@ -0,0 +1,39 @@
+﻿# Copyright 2011 - Present RealDimensions Software, LLC & original authors/contributors from https://github.com/chocolatey/chocolatey
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function Generate-BinFile {
+  param(
+    [string] $name, 
+    [string] $path,
+    [string] $outPath,
+    [switch] $useStart
+  )
+  
+  Write-Debug "Running 'Generate-BinFile' for $name with path:'$path'";
+
+  $packageBatchFileName = Join-Path $outPath "$name.bat"
+  $path  = Join-Path $path "$name.exe"
+  Write-Host "Adding $packageBatchFileName and pointing to '$path'."
+  
+  if ($useStart) {
+    Write-Host "Setting up $name as a non-command line application."
+    "@echo off
+    SET DIR=%~dp0%
+    start """" ""$path"" %*" | Out-File $packageBatchFileName -encoding ASCII     
+  } else {
+    "@echo off
+    SET DIR=%~dp0%
+    ""$path"" %*" | Out-File $packageBatchFileName -encoding ASCII     
+  }
+}
```

Comment:
This should probably look closer to what the last version of POSH chocolatey looked like here, where it fell back to this if shimgen failed.